sw/device/sca/sha3_serial.c

```diff
@@ -216,14 +256,16 @@ static dif_result_t kmac_msg_start(dif_kmac_mode_kmac_t mode, size_t l,
   }
 
   // Set key length and shares.
+  uint32_t a;
   mmio_region_write32(kmac.base_addr, KMAC_KEY_LEN_REG_OFFSET, key_len);
   for (int i = 0; i < ARRAYSIZE(k->share0); ++i) {
+    a = next_lfsr();
     mmio_region_write32(kmac.base_addr,
                         KMAC_KEY_SHARE0_0_REG_OFFSET + i * sizeof(uint32_t),
-                        k->share0[i]);
+                        k->share0[i]^a);
     mmio_region_write32(kmac.base_addr,
                         KMAC_KEY_SHARE1_0_REG_OFFSET + i * sizeof(uint32_t),
-                        k->share1[i]);
+                        k->share1[i]^a);
```

Comment:
`a` is generated with the `next_lfsr` function, which is documented to be a PRNG for randomizing the sampling order during batch capture. But I think these lines use `a` to mask the shares of the key, right? Does that work? If so, could you add a comment why this works?